src/containers/barray.mli

```diff
@@ -252,100 +216,71 @@ val bsearch : ?cmp:('a -> 'a -> int) -> 'a -> 'a t ->
     - [`Just_after i] if [arr.(i) < x < arr.(i+1)]
     - [`Empty] if the array is empty
 
-    @raise Invalid_argument if the array is found to be unsorted w.r.t [cmp]
-    @since 0.13 *)
-
-val for_all : ('a -> bool) -> 'a t -> bool
+    @raise Invalid_argument if the array is found to be unsorted w.r.t [cmp] *)
 
-val for_all2 : ('a -> 'b -> bool) -> 'a t -> 'b t -> bool
+val forAll2 : ('a -> 'b -> bool) -> 'a t -> 'b t -> bool
 (** Forall on pairs of arrays.
     @raise Invalid_argument if they have distinct lengths
-    allow different types @since 0.20 *)
-
-val exists : ('a -> bool) -> 'a t -> bool
+    allow different types *)
 
 val exists2 : ('a -> 'b -> bool) -> 'a t -> 'b t -> bool
 (** Exists on pairs of arrays.
     @raise Invalid_argument if they have distinct lengths
-    allow different types @since 0.20 *)
+    allow different types *)
 
 val fold2 : ('acc -> 'a -> 'b -> 'acc) -> 'acc -> 'a t -> 'b t -> 'acc
 (** Fold on two arrays stepwise.
-    @raise Invalid_argument if they have distinct lengths
-    @since 0.20 *)
+    @raise Invalid_argument if they have distinct lengths *)
 
 val iter2 : ('a -> 'b -> unit) -> 'a t -> 'b t -> unit
 (** Iterate on two arrays stepwise.
-    @raise Invalid_argument if they have distinct lengths
-    @since 0.20 *)
+    @raise Invalid_argument if they have distinct lengths *)
 
 val shuffle : 'a t -> unit
 (** Shuffle randomly the array, in place *)
 
-val shuffle_with : Random.State.t -> 'a t -> unit
+val shuffleWith : Random.State.t -> 'a t -> unit
 (** Like shuffle but using a specialized random state *)
 
-val random_choose : 'a t -> 'a random_gen
-(** Choose an element randomly.
-    @raise Not_found if the array/slice is empty *)
-
-val to_seq : 'a t -> 'a sequence
-val to_gen : 'a t -> 'a gen
-val to_klist : 'a t -> 'a klist
-
-(** {2 IO} *)
-
-val pp: ?sep:string -> 'a printer -> 'a t printer
-(** Print an array of items with printing function *)
-
-val pp_i: ?sep:string -> (int -> 'a printer) -> 'a t printer
-(** Print an array, giving the printing function both index and item *)
+val toIterator: 'a t -> 'a Iterator.t
```

Comment:
An iterator is really just a "view" of or interface to the data structure, not itself a separate data structure, so I don't think `toIterator` is a good name for this. `getIterator` or `makeIterator`, or perhaps just `iterator` would be more fitting I think.